src/State.ts

```diff
@@ -98,13 +104,33 @@ export function versionControl() {
       for (const t of newStatues) state.value.chars[key].statues[t] = 0;
     }
   }
+  // Add new bubble slots and a goals field for each bubble
+  if (state.value.version < "0.2.4") {
+    let colors: Color[] = ["Orange", "Green", "Purple", "Yellow"];
+    for (const k of colors) {
+      let amount = 15;
+      if (!state.value.alchemy.goals) {
+        state.value.alchemy.goals = {
+          Orange: [],
+          Green: [],
+          Purple: [],
+          Yellow: [],
+        };
+      }
+      for (let i = 0; i < amount; i++) {
+        state.value.alchemy.upgrades[k][i] = state.value.alchemy.upgrades[k][i] ?? 0; 
```

Comment:
(nit) I think alchemy is already handled in version control so this might be redundant.